### PR TITLE
Documentation Updates

### PR DESCRIPTION
**Pull request by plun1331**
## Summary

Changes and fixes a lot of stuff in the documentation.

Things I remember doing:
- Documented cogs
- Documented ApplicationCommand exceptions and moved extension errors to `discord` from `ext.commands`
- Fixed many typos
- Added guide to index
- Renamed migrating to migrating to v1 to make room for a potential v2 migration guide
- Redid half the sink docs
- Fixed warnings while building docs
- Fixed a lot of broken references
- Notes the message content intent requirement at the top of ext.commands documentation (could be moved?)

**You can see a preview at https://pycord-plun1331.readthedocs.io/en/latest/index.html**

## Checklist

- [ ] If code changes were made then they have been tested.
    - [ ] I have updated the documentation to reflect the changes.
- [ ] If `type: ignore` comments were used, a comment is also left explaining why
- [ ] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [x] This PR is **not** a code change (e.g. documentation, README, typehinting, examples, ...)

### CODE REVIEW

File: discord/sinks/core.py [hunk 1]
```diff
@@ -118,20 +118,22 @@ def __init__(self, data, client):
...
-        The AudioData is still writing
+    .. versionadded:: 2.0
     """
 
     def __init__(self, file):
         self.file = file
         self.finished = False
 
     def write(self, data):
+        """
```
Comment (ooliver1) [inline discord/sinks/core.py:129]: may i suggest using the [pep 257 convention](https://peps.python.org/pep-0257) of docstrings
```py
"""short

longer
description
"""
```

File: docs/api.rst [hunk 2]
```diff
@@ -103,6 +103,10 @@ ApplicationCommand
 
 .. autoclass:: ApplicationCommand
     :members:
+    
+.. autofunction:: discord.commands.core.application_command
```
Comment (ooliver1) [inline docs/api.rst:107]: is this a decorator? you may want the `:decorator:` option from sphinxcontrib-trio

File: docs/conf.py [hunk 3]
```diff
@@ -174,6 +174,7 @@
     "issues": "https://github.com/Pycord-Development/pycord/issues",
     "discussions": "https://github.com/Pycord-Development/pycord/discussions",
     "examples": f"https://github.com/Pycord-Development/pycord/tree/{branch}/examples",
+    "guide": "https://guide.pycord.dev/"
```
Comment (ooliver1) [inline docs/conf.py:177]: trailing commas make sense for VCS so git diffs are only one line long, like how it is right now

**Review by ooliver1 — CHANGES_REQUESTED** [review]

**Comment by ooliver1** [comment]
oh also id highly recommend using proper commit messages when contributing - or just any time you make a commit message
helps with seeing what each commit did and when

**Comment by plun1331** [comment]
Suggested changes have been applied

**Review by Dorukyum — APPROVED** [review]
Looks good to me, thank you

**Review by Middledot — APPROVED** [review]
lgtm

**Review by krittick — APPROVED** [review]

**Pull request merged**